src/ucp/tag/offload.c

```diff
@@ -556,9 +558,17 @@ ucs_status_t ucp_tag_offload_sw_rndv(uct_pending_req_t *self)
     rndv_rts_hdr = ucs_alloca(rndv_hdr_len);
     packed_len   = ucp_tag_rndv_rts_pack(rndv_rts_hdr, req);
 
-    return uct_ep_tag_rndv_request(ep->uct_eps[req->send.lane],
-                                   req->send.msg_proto.tag.tag,
-                                   rndv_rts_hdr, packed_len, 0);
+    ucs_assert((rndv_rts_hdr->address != 0) || !UCP_DT_IS_CONTIG(req->send.datatype) ||
+               !ucp_rndv_is_get_zcopy(req, ep->worker->context));
+    status = uct_ep_tag_rndv_request(ep->uct_eps[req->send.lane],
+                                     req->send.msg_proto.tag.tag,
+                                     rndv_rts_hdr, packed_len, 0);
+    if (ucs_unlikely(status != UCS_OK)) {
+        return status;
+    }
+
+    ucp_request_send_track(req);
```

Comment:
why needed?

Reply:
since, we will receive ATS and we will untrack the request